pkg/utils/cmp.go

```diff
@@ -5,44 +5,51 @@ import (
 	"strings"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/infracloudio/botkube/pkg/config"
 )
 
-// SpecDiffReporter is a simple custom reporter that only records differences
-// detected in Object Spec during comparison.
-type SpecDiffReporter struct {
+// DiffReporter is a simple custom reporter that records differences
+// detected in Object during comparison.
+type DiffReporter struct {
+	field string
 	path  cmp.Path
 	diffs []string
 }
 
 // PushStep custom implements Reporter interface
-func (r *SpecDiffReporter) PushStep(ps cmp.PathStep) {
+func (r *DiffReporter) PushStep(ps cmp.PathStep) {
 	r.path = append(r.path, ps)
 }
 
 // Report custom implements Reporter interface
-func (r *SpecDiffReporter) Report(rs cmp.Result) {
+func (r *DiffReporter) Report(rs cmp.Result) {
 	if !rs.Equal() {
 		vx, vy := r.path.Last().Values()
 		path := fmt.Sprintf("%#v", r.path)
-		if ok := strings.Contains(path, ".Spec."); ok {
+		if ok := strings.Contains(path, "."+strings.Title(r.field)); ok {
```

Comment:
I think we should support regex/jsonpath here.
Suppose I am interested only in change in container status of Pod, so I can just set `status.containerStatuses[*].ready` in updateSetting and that should work.

Reply:
I have some working code changes for jsonpath on my local repo. I'll cherry-pick the commits and  raise a new PR